### PR TITLE
Implement internal/errors/blob test

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:
This PR implements internal/errors/blob test

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/errors/blob_test.go [hunk 1]
```diff
@@ -313,3 +267,125 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
+		if !Is(err, w.err) {
+			return Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			err := New("internal error")
+			return test{
+				name: "unwrap returns the unwrapped error",
```
Comment (hlts2) [inline internal/errors/blob_test.go:357]:
```suggestion
				name: "returns the unwrapped error",
```

File: internal/errors/blob_test.go [hunk 2]
```diff
@@ -313,3 +267,125 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
```
Comment (hlts2) [inline internal/errors/blob_test.go:296]:
```suggestion
				name: "returns the unwrapped error",
```

File: internal/errors/blob_test.go [hunk 3]
```diff
@@ -191,31 +168,15 @@ func TestErrBlobNoSuchKey_Error(t *testing.T) {
...
-		           fields: fields {
-		           err: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "error return error string",
```
Comment (hlts2) [inline internal/errors/blob_test.go:172]:
```suggestion
			name: "returns error string",
```

File: internal/errors/blob_test.go [hunk 4]
```diff
@@ -265,31 +226,24 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
+			name: "return true when the error is ErrBlobNoSuchKey",
+			args: args{
+				err: &ErrBlobNoSuchKey{New("nosuchkeyerr")},
+			},
+			want: want{
+				want: true,
+			},
+		},
+		{
+			name: "return false when the error is not ErrBlobNoSuchKey",
```
Comment (hlts2) [inline internal/errors/blob_test.go:239]:
```suggestion
			name: "returns false when the error is not ErrBlobNoSuchKey",
```

File: internal/errors/blob_test.go [hunk 5]
```diff
@@ -265,31 +226,24 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
-		           args: args {
-		           err: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "return true when the error is ErrBlobNoSuchKey",
```
Comment (hlts2) [inline internal/errors/blob_test.go:230]:
```suggestion
			name: "returns true when the error is ErrBlobNoSuchKey",
```

File: internal/errors/blob_test.go [hunk 6]
```diff
@@ -120,31 +104,24 @@ func TestIsErrBlobNoSuchBucket(t *testing.T) {
...
-		           args: args {
-		           err: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "return true when the error is ErrBlobNoSuchBucket",
```
Comment (hlts2) [inline internal/errors/blob_test.go:108]:
```suggestion
			name: "returns true when the error is ErrBlobNoSuchBucket",
```

File: internal/errors/blob_test.go [hunk 7]
```diff
@@ -120,31 +104,24 @@ func TestIsErrBlobNoSuchBucket(t *testing.T) {
...
+			args: args{
+				err: &ErrBlobNoSuchBucket{New("nosuchbucketerr")},
+			},
+			want: want{
+				want: true,
+			},
+		},
+		{
+			name: "return false when the error is not ErrBlobNoSuchBucket",
```
Comment (hlts2) [inline internal/errors/blob_test.go:117]:
```suggestion
			name: "returns false when the error is not ErrBlobNoSuchBucket",
```

File: internal/errors/blob_test.go [hunk 8]
```diff
@@ -46,31 +46,15 @@ func TestErrBlobNoSuchBucket_Error(t *testing.T) {
...
-		           fields: fields {
-		           err: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "error return error string",
```
Comment (hlts2) [inline internal/errors/blob_test.go:50]:
```suggestion
			name: "returns error string",
```

File: internal/errors/blob.go [hunk 9]
```diff
@@ -18,49 +18,58 @@
...
 	target := new(ErrBlobNoSuchBucket)
 	return As(err, &target)
 }
 
+// ErrBlobNoSuchKey represents no such key error of S3.
 type ErrBlobNoSuchKey struct {
 	err error
 }
 
+// Error return the string representation of the internal error.
```
Comment (hlts2) [inline internal/errors/blob.go:62]:
```suggestion
// Error returns the string representation of the internal error.
```

File: internal/errors/blob_test.go [hunk 10]
```diff
@@ -265,31 +226,24 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
+		},
+		{
+			name: "return false when the error is not ErrBlobNoSuchKey",
+			args: args{
+				err: New("anothererr"),
+			},
+			want: want{
+				want: false,
+			},
+		},
```
Comment (hlts2) [inline internal/errors/blob_test.go:246]: Should we add a test case when error is nil?

File: internal/errors/blob_test.go [hunk 11]
```diff
@@ -120,31 +104,24 @@ func TestIsErrBlobNoSuchBucket(t *testing.T) {
...
+		{
+			name: "return false when the error is not ErrBlobNoSuchBucket",
+			args: args{
+				err: New("anothererr"),
+			},
+			want: want{
+				want: false,
+			},
+		},
```
Comment (hlts2) [inline internal/errors/blob_test.go:124]: Should we add test case when err is nil ?

**Review by hlts2 — CHANGES_REQUESTED** [review]

File: internal/errors/blob_test.go [hunk 12]
```diff
@@ -120,31 +104,33 @@ func TestIsErrBlobNoSuchBucket(t *testing.T) {
...
+				err: &ErrBlobNoSuchBucket{New("nosuchbucketerr")},
+			},
+			want: want{
+				want: true,
+			},
+		},
+		{
+			name: "returns false when the error is not ErrBlobNoSuchBucket",
+			args: args{
+				err: New("anothererr"),
```
Comment (vankichi) [inline internal/errors/blob_test.go:119]:
```suggestion
				err: &ErrBlobNoSuchKey{New("no such key err")}
```
how about it?
i think it'd be more meaningful 🤔

File: internal/errors/blob_test.go [hunk 13]
```diff
@@ -265,31 +235,33 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
+				err: &ErrBlobNoSuchKey{New("nosuchkeyerr")},
+			},
+			want: want{
+				want: true,
+			},
+		},
+		{
+			name: "returns false when the error is not ErrBlobNoSuchKey",
+			args: args{
+				err: New("anothererr"),
```
Comment (vankichi) [inline internal/errors/blob_test.go:250]:
```suggestion
				err: &ErrBlobNoSuchBucket{New("no such bucket err")},
```
same 🤔

File: internal/errors/blob_test.go [hunk 14]
```diff
@@ -313,3 +285,143 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
+				fields: fields{
+					err: err,
+				},
+				want: want{
+					err: err,
+				},
+			}
+		}(),
+		{
+			name: "returns the unwrapped nil error",
```
Comment (vankichi) [inline internal/errors/blob_test.go:324]:
```suggestion
			name: "returns nil when get the unwrapped nil error",
```

File: internal/errors/blob_test.go [hunk 15]
```diff
@@ -313,3 +285,143 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
```
Comment (vankichi) [inline internal/errors/blob_test.go:394]:
```suggestion
			name: "returns nil when get the unwrapped nil error",
```

File: internal/errors/blob_test.go (hunk 14 above)
Comment (kevindiu) [reply internal/errors/blob_test.go:324]: I think it is more correct.
```suggestion
			name: "returns nil when getting the unwrapped nil error",
```

File: internal/errors/blob_test.go (hunk 15 above)
Comment (kevindiu) [reply internal/errors/blob_test.go:394]:
```suggestion
			name: "returns nil when getting the unwrapped nil error",
```

**Review by hlts2 — APPROVED** [review]
LGTM

File: internal/errors/blob_test.go [hunk 16]
```diff
@@ -46,31 +46,15 @@ func TestErrBlobNoSuchBucket_Error(t *testing.T) {
...
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "returns error string",
+			fields: fields{
+				err: New("err"),
```
Comment (vankichi) [inline internal/errors/blob_test.go:52]:
```suggestion
				err: New("err: no such bucket"),
```

File: internal/errors/blob_test.go [hunk 17]
```diff
@@ -265,31 +235,33 @@ func TestIsErrBlobNoSuchKey(t *testing.T) {
...
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "returns true when the error is ErrBlobNoSuchKey",
+			args: args{
+				err: &ErrBlobNoSuchKey{New("nosuchkeyerr")},
```
Comment (vankichi) [inline internal/errors/blob_test.go:241]:
```suggestion
				err: &ErrBlobNoSuchKey{New("no such key err")},
```

File: internal/errors/blob_test.go [hunk 18]
```diff
@@ -120,31 +104,33 @@ func TestIsErrBlobNoSuchBucket(t *testing.T) {
...
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "returns true when the error is ErrBlobNoSuchBucket",
+			args: args{
+				err: &ErrBlobNoSuchBucket{New("nosuchbucketerr")},
```
Comment (vankichi) [inline internal/errors/blob_test.go:110]:
```suggestion
				err: &ErrBlobNoSuchBucket{New("no such bucket err")},
```

File: internal/errors/blob_test.go [hunk 19]
```diff
@@ -46,31 +46,15 @@ func TestErrBlobNoSuchBucket_Error(t *testing.T) {
...
-		       }
-		   }(),
-		*/
+		{
+			name: "returns error string",
+			fields: fields{
+				err: New("err"),
+			},
+			want: want{
+				want: "err",
```
Comment (kevindiu) [inline internal/errors/blob_test.go:55]:
```suggestion
				want: "err: no such bucket",
```

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]
LGTM
